backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java

```diff
@@ -112,8 +113,20 @@ public Conversation editTitle(Conversation conversation, String title) throws Un
             throw new UnAuthorizedException();
         }
 
-        return conversationPersistenceManager.save(
-                conversation.toBuilder().title(title).build(), user.get());
+        return getConversation(conversation, title, user);
+    }
+
+    private Conversation getConversation(Conversation conversation, String title, Optional<User> user) {
+        conversation = Conversation.builder()
+                .id(conversation.getId())
+                .discussions(conversation.getDiscussions())
+                .title((conversationPersistenceManager.save(
```

Comment:
persistence save being called here can be misleading since the function name is `getConversation` and we are saving a title inside , persistence save should not be here

Reply:
helper method is added